jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/Fields.java

```diff
@@ -35,65 +35,59 @@
  */
 public class Fields implements Iterable<Fields.Field>
 {
-    private final boolean caseSensitive;
+    public static final Fields EMPTY = new Fields(Collections.emptyMap());
+
     private final Map<String, Field> fields;
 
     /**
      * <p>Creates an empty, modifiable, case insensitive {@code Fields} instance.</p>
-     *
-     * @see #Fields(Fields, boolean)
      */
     public Fields()
     {
         this(false);
     }
 
     /**
-     * <p>Creates an empty, modifiable, case insensitive {@code Fields} instance.</p>
+     * <p>Creates an empty, modifiable, case insensitive {@code Fields} instance.</p>**
      *
      * @param caseSensitive whether this {@code Fields} instance must be case sensitive
-     * @see #Fields(Fields, boolean)
      */
     public Fields(boolean caseSensitive)
     {
-        this.caseSensitive = caseSensitive;
-        fields = new LinkedHashMap<>();
+        this(caseSensitive ? new LinkedHashMap<>() : new TreeMap<>(String::compareToIgnoreCase));
     }
 
-    /**
-     * <p>Creates a {@code Fields} instance by copying the fields from the given
-     * {@code Fields} and making it (im)mutable depending on the given {@code immutable} parameter</p>
-     *
-     * @param original the {@code Fields} to copy fields from
-     * @param immutable whether this instance is immutable
-     */
-    public Fields(Fields original, boolean immutable)
+    public Fields(Map<String, Field> fields)
     {
-        this.caseSensitive = original.caseSensitive;
-        Map<String, Field> copy = new LinkedHashMap<>(original.fields);
-        fields = immutable ? Collections.unmodifiableMap(copy) : copy;
+        this.fields = fields;
+    }
+
+    public Fields asImmutable()
+    {
+        return new Fields(Collections.unmodifiableMap(fields));
     }
 
     @Override
     public boolean equals(Object obj)
     {
         if (this == obj)
             return true;
-        if (obj == null || getClass() != obj.getClass())
-            return false;
-        Fields that = (Fields)obj;
-        if (getSize() != that.getSize())
-            return false;
-        if (caseSensitive != that.caseSensitive)
+        if (obj == null)
             return false;
-        for (Map.Entry<String, Field> entry : fields.entrySet())
+        if (obj instanceof Fields that)
         {
-            String name = entry.getKey();
-            Field value = entry.getValue();
-            if (!value.equals(that.get(name), caseSensitive))
+            if (getSize() != that.getSize())
                 return false;
+            for (Map.Entry<String, Field> entry : fields.entrySet())
+            {
+                String name = entry.getKey();
+                Field value = entry.getValue();
+                if (!value.equals(that.get(name)))
+                    return false;
+            }
+            return true;
```

Comment:
perhaps.  But there is the case insensitivity of field names, but not values that comes into play.   So I'm happy with small changes to existing code rather than replacement